docs/conf.py

```diff
@@ -65,6 +70,46 @@
 # (note that `group` is the GitHub user or organization)
 issues_github_path = "skops-dev/skops"
 
+
+def linkcode_resolve(domain, info):
+    if domain != "py":
+        return None
+    if not info["module"]:
+        return None
+    filename = info["module"].replace(".", "/")
+    if domain not in ("py", "pyx"):
+        return
+    if not info.get("module") or not info.get("fullname"):
+        return
+
+    class_name = info["fullname"].split(".")[0]
+    module = __import__(info["module"], fromlist=[class_name])
+    obj = attrgetter(info["fullname"])(module)
+
+    # Unwrap the object to get the correct source
+    # file in case that is wrapped by a decorator
+    obj = inspect.unwrap(obj)
+
+    try:
+        fn = inspect.getsourcefile(inspect.unwrap(obj))
+    except TypeError:
+        try: 
+            fn = inspect.getsourcefile(inspect.unwrap(obj.fget))
+        except (AttributeError, TypeError):
+            fn = None
+    if not fn:
+        return None
+    package = "skops"
+    fn = os.path.relpath(fn, start=os.path.dirname(__import__(package).__file__))
+    try:
+        lineno = inspect.getsourcelines(obj)[1]
+    except Exception:
+        lineno = ""
+    return (
+        "https://github.com/skops-dev/skops/blob/main/skops/" + fn + "#L" + str(lineno)
```

Comment:
this shouldn't refer to the `main` branch, it should refer to the branch for which this documentation is being built. For example, in sklearn in [this page](https://scikit-learn.org/stable/modules/generated/sklearn.base.ClassifierMixin.html#sklearn.base.ClassifierMixin) for `ClassifierMixin` you get [this link](https://github.com/scikit-learn/scikit-learn/blob/36958fb24/sklearn/base.py#L635) which is not on `main`.

Reply:
Could you try this patch:

``` patch
diff --git a/docs/conf.py b/docs/conf.py
index 3ed773f..4ed5d3d 100644
--- a/docs/conf.py
+++ b/docs/conf.py
@@ -8,6 +8,7 @@
 
 import inspect
 import os
+import subprocess
 from operator import attrgetter
 
 from packaging.version import parse
@@ -71,6 +72,19 @@ autosummary_generate = True
 issues_github_path = "skops-dev/skops"
 
 
+# -- linkcode configuration --------------------------------------------------
+REVISION_CMD = "git rev-parse --short HEAD"
+
+
+def _get_git_revision():
+    try:
+        revision = subprocess.check_output(REVISION_CMD.split()).strip()
+    except (subprocess.CalledProcessError, OSError):
+        print("Failed to execute git to get revision")
+        return None
+    return revision.decode("utf-8")
+
+
 def linkcode_resolve(domain, info):
     if domain != "py":
         return None
@@ -93,7 +107,7 @@ def linkcode_resolve(domain, info):
     try:
         fn = inspect.getsourcefile(inspect.unwrap(obj))
     except TypeError:
-        try: 
+        try:
             fn = inspect.getsourcefile(inspect.unwrap(obj.fget))
         except (AttributeError, TypeError):
             fn = None
@@ -105,9 +119,16 @@ def linkcode_resolve(domain, info):
         lineno = inspect.getsourcelines(obj)[1]
     except Exception:
         lineno = ""
-    return (
-        "https://github.com/skops-dev/skops/blob/main/skops/" + fn + "#L" + str(lineno)
+    url_fmt = (
+        "https://github.com/scikit-learn/"
+        "scikit-learn/blob/{revision}/"
+        "{package}/{path}#L{lineno}"
     )
+    revision = _get_git_revision()
+    return url_fmt.format(revision=revision, package=package, path=fn, lineno=lineno)
+
+
+# -- linkcode configuration --------------------------------------------------
 
 
 # -- Options for HTML output -------------------------------------------------
```

Reply:
Returns something like this: https://github.com/scikit-learn/scikit-learn/blob/a6db88f/skops/hub_utils/_hf_hub.py#L594 
and also doesn't work

Reply:
Ahahahahhahahhahah I feel like an idiot. Thank you so much!